python/pyspark/sql/column.py

```diff
@@ -175,8 +175,15 @@ def __init__(self, jc):
     _eqNullSafe_doc = """
     Equality test that is safe for null values.
 
-    :param other: a value or :class:`Column`
+    .. versionadded:: 2.3.0
+
+    Parameters
+    ----------
+    other
```

Comment:
I simply made it un-typed for simplicity where it takes a column + any value that a literal supports.